pkg/storage/gcp/checker.go

```diff
@@ -2,20 +2,11 @@ package gcp
 
 import (
 	"context"
-
-	"github.com/gomods/athens/pkg/config"
 )
 
 // Exists implements the (./pkg/storage).Checker interface
 // returning true if the module at version exists in storage
 func (s *Storage) Exists(module, version string) bool {
 	ctx := context.Background()
```

Comment:
should the `storage.Backend` interface take a `context.Context`  in all of its methods? They are all remote calls by default so imo they should. Otherwise, we should at least do a `context.WithTimeout` instead of Background. Doesn't have to be in this pr.

Reply:
That's a great call. We should open an issue regarding the timeout. Not so backends are network. Most but not file or memory. However we do intend to thread context down the whole stack so it can't hurt to start somewhere that makes sense. Then just throw a timeout on the context before passing it to one in the backend calls. Keep in mind that the file uploads already share a default timeout of five minutes.

Reply:
For sure another PR though